crates/scheduler/src/call/crypto/hash.rs

```diff
@@ -156,14 +169,17 @@ fn hmac_update<B: Board>(mut call: SchedulerCall<B, api::hmac_update::Sig>) {
     let memory = scheduler.applet.store.memory();
     let result = try {
         let data = memory.get(*data, *length)?;
-        match scheduler.applet.hashes.get_mut(*id as usize)? {
+        let context = match scheduler.applet.hashes.take(*id as usize)? {
```

Comment:
It shouldn't be needed to `take` here.

Reply:
With `get_mut`, the `hash` in `HashContext::Sha256(hash) => HashApi(hash).update(data)` is `&mut`, and it couldn't be moved.